packages/core/strapi/lib/services/entity-service/types/params/index.d.ts

```diff
@@ -2,10 +2,11 @@ import type { Common } from '@strapi/strapi';
 
 import type * as Sort from './sort';
 import type * as Pagination from './pagination';
+import type * as Fields from './fields';
 
 export type For<TSchemaUID extends Common.UID.Schema> = {
   sort?: Sort.Any<TSchemaUID>;
-  pagination?: Pagination.Any;
-};
+  fields?: Fields.Any<TSchemaUID>;
+} & Pagination.Any;
```

Comment:
How come? Doesn't it have to be inside the For types as it was? `pagination?: Pagination.Any` ?

Reply:
No, pagination options are at the Root level for the params